server/storage/sqldb.go

```diff
@@ -45,13 +45,44 @@ func translateOldVersionError(err error) error {
 	return err
 }
 
+// TufFilesInChannels scopes gorm queries to a particular channel or set of channels
+func TufFilesInChannels(channels ...*Channel) func(db *gorm.DB) *gorm.DB {
+	channelIDs := []uint{}
+	for _, channel := range channels {
+		channelIDs = append(channelIDs, channel.ID)
+	}
+
+	// If no channels were added, check Published
+	if len(channelIDs) == 0 {
+		channelIDs = []uint{Published.ID}
+	}
+
+	// If only one channel is provided, we don't need to group/collect
+	if len(channelIDs) == 1 {
+		return func(db *gorm.DB) *gorm.DB {
+			return db.Joins("INNER JOIN channels_tuf_files ON tuf_files.id = channels_tuf_files.tuf_file_id").
+				Where("channel_id = ?", channelIDs[0])
+		}
+	}
+
+	// Find tuf files matching all passed channels
+	return func(db *gorm.DB) *gorm.DB {
+		return db.Joins("INNER JOIN channels_tuf_files ON tuf_files.id = channels_tuf_files.tuf_file_id").
+			Where("channel_id IN (?)", channelIDs).
```

Comment:
I am likely just misunderstanding how this works, but would this only enforce that the rows were in the right number of channels, as opposed to the channels specified?  For instance, if there were 3 channels (`Alice`, `Bob`, and `Carol`), and I wanted to get the latest metadata for `GUN` and `Timestamp` that was in both `Alice` and `Carol`'s channels, but the database had:

1. id=`5`, `GUN`, `Timestamp`, channel=`Alice` (via the `channel_tuf_files` table)
1. id=`5`, `GUN`,  `Timestamp`, channel=`Bob` (via the `channel_tuf_files` table)

Would the scope consider that good because the `id=5` is in one of (`Alice`, `Carol`), and it it is also in 2 distinct channels (`Alice`, `Bob`)?